### PR TITLE
[M05] Missing counter addition

### PR DESCRIPTION
**Pull request by mrsmkl**
### Related issues

- Fixes https://github.com/celo-org/celo-labs/issues/268

### CODE REVIEW

**Comment by nambrot** [comment]
Can we set the number to something higher? Is 100 ok?

File: packages/protocol/contracts/identity/Attestations.sol [hunk 1]
```diff
@@ -550,6 +575,23 @@ contract Attestations is
     return (matches, new address[](totalAddresses));
   }
 
+  function getPossibleAttestationIssuers() public view returns (address[] memory, uint256) {
```
Comment (nambrot) [inline packages/protocol/contracts/identity/Attestations.sol]: Was this extraction necessary? It also contains an error as it makes it possible to add the same issuer more than once?

File: packages/protocol/contracts/identity/Attestations.sol [hunk 2]
```diff
@@ -562,35 +604,29 @@ contract Attestations is
...
-        continue;
-      }
-
       currentIndex = currentIndex.add(1);
       attestation.status = AttestationStatus.Incomplete;
       attestation.blockNumber = unselectedRequest.blockNumber;
       attestation.attestationRequestFeeToken = unselectedRequest.attestationRequestFeeToken;
       state.selectedIssuers.push(issuer);
 
+      issuersLength = issuersLength.sub(1);
```
Comment (nambrot) [inline packages/protocol/contracts/identity/Attestations.sol:644]: I don't understand this logic, it just pops it out from the end of the list?

**Review by nambrot — CHANGES_REQUESTED** [review]

File: packages/protocol/contracts/identity/Attestations.sol (hunk 1 above)
Comment (mrsmkl) [reply packages/protocol/contracts/identity/Attestations.sol]: Yeah, it's more efficient to not extract everything at start. Added a test to check that validator cannot be selected twice.

File: packages/protocol/contracts/identity/Attestations.sol (hunk 2 above)
Comment (mrsmkl) [reply packages/protocol/contracts/identity/Attestations.sol:644]: Added comment, hopefully that helps.

**Review by nambrot — APPROVED** [review]

**Pull request merged**